octoprint_mrbeam/software_update_information.py

```diff
@@ -1,69 +1,178 @@
-from datetime import datetime, date
+import json
 import os, sys
+import threading
+from datetime import date, datetime
+from os.path import join
 
-from octoprint.util import dict_merge
-from octoprint_mrbeam import IS_X86
+import requests
+
+from octoprint_mrbeam import MrBeamEvents
 from octoprint_mrbeam.mrb_logger import mrb_logger
-from octoprint_mrbeam.util import logExceptions
+from octoprint_mrbeam.util.log import logme
 from util.pip_util import get_version_of_pip_module
 
-
 SW_UPDATE_TIER_PROD = "PROD"
-SW_UPDATE_TIER_BETA = "BETA"
 SW_UPDATE_TIER_DEV = "DEV"
+SW_UPDATE_TIER_BETA = "BETA"
 DEFAULT_REPO_BRANCH_ID = {
     SW_UPDATE_TIER_PROD: "stable",
     SW_UPDATE_TIER_BETA: "beta",
     SW_UPDATE_TIER_DEV: "develop",
 }
 
+SW_UPDATE_FILE = "update_info.json"
+SW_UPDATE_CLOUD_PATH = "https://mr-beam.org/beamos/config/sw-update-conf"
+
 # add to the display name to modules that should be shown at the top of the list
 SORT_UP_PREFIX = " "
 
-
 _logger = mrb_logger("octoprint.plugins.mrbeam.software_update_information")
 
+sw_update_config = dict()
+
 # Commented constants are kept in case we update more packages from the virtualenv
 # GLOBAL_PY_BIN = "/usr/bin/python2.7"
 # VENV_PY_BIN = sys.executable
 GLOBAL_PIP_BIN = "/usr/local/bin/pip"
 GLOBAL_PIP_COMMAND = (
     "sudo {}".format(GLOBAL_PIP_BIN) if os.path.isfile(GLOBAL_PIP_BIN) else None
 )
-# GLOBAL_PIP_COMMAND = "sudo {} -m pip".format(GLOBAL_PY_BIN) if os.path.isfile(GLOBAL_PY_BIN) else None #  --disable-pip-version-check
-# VENV_PIP_COMMAND = ("%s -m pip --disable-pip-version-check" % VENV_PY_BIN).split(' ') if os.path.isfile(VENV_PY_BIN) else None
+
 BEAMOS_LEGACY_DATE = date(2018, 1, 12)
 
+_softwareupdate_handler = None
 
-def get_update_information(plugin):
-    result = dict()
 
+def get_modules():
+    return sw_update_config
+
+
+class MrBeamSoftwareupdateHandler:
+    def __init__(self, plugin, tier, beamos_date):
+        self._plugin = plugin
+        self._tier = tier
+        self._beamos_date = beamos_date
```

Comment:
Why is this class necessary?
It does not seem like you are using it's attributes, and `self` is only used for referencing `self._plugin`, so to me it looks redundant. I would suggest to keep this on a functional approach instead of being Object Oriented.

Reply:
hmm, I can't remember why I choose to do it Object oriented, you are right I only use the self._plugin
What is the downside to do it this way?